.github/workflows/test.yml

```diff
@@ -226,14 +226,63 @@ jobs:
           name: generated_binding-${{ matrix.config.os }}-R-${{ matrix.config.r }}-rust-${{ matrix.config.rust-version }}-${{ matrix.config.target || 'default'}}
           path: generated_bindings
 
-      # Run tests again using different bindings
-      - name: Run tests on precomputed bindings shipped with libR-sys
-        if: matrix.config.skip-tests != 'true'
-        run: |
-          . ./ci-cargo.ps1
-          ci-cargo test -vv $(if ($env:RUST_TARGET -ne '') {"--target=$env:RUST_TARGET"} ) '--' --nocapture -ActionName "Running tests for target: $env:RUST_TARGET"
-        env:
-          RUST_TARGET: ${{ matrix.config.target }}
+      # # Run tests again using different bindings
+      # - name: Run tests on precomputed bindings shipped with libR-sys
+      #   if: matrix.config.skip-tests != 'true'
+      #   run: |
+      #     . ./ci-cargo.ps1
+      #     ci-cargo test -vv $(if ($env:RUST_TARGET -ne '') {"--target=$env:RUST_TARGET"} ) '--' --nocapture -ActionName "Running tests for target: $env:RUST_TARGET"
+      #   env:
+      #     RUST_TARGET: ${{ matrix.config.target }}
+  check_generate_bindings_flag:
+    name: Check if [generate bindings] is in latest commit message
+    runs-on: ubuntu-latest
+    outputs:
+      head_commit_message: ${{ steps.get_head_commit_message.outputs.HEAD_COMMIT_MESSAGE }}
+      # generate_bindings: ${{ contains(steps.get_head_commit_message.outputs.HEAD_COMMIT_MESSAGE, '[generate bindings]') }}
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.ref }}
+    - name: Get Head Commit Message
+      id: get_head_commit_message
+      run: echo "HEAD_COMMIT_MESSAGE=$(git show -s --format=%s)" >> "$GITHUB_OUTPUT"
+    - name: Show commit message
+      run: |
+        echo "${{ steps.get_head_commit_message.outputs.HEAD_COMMIT_MESSAGE }}"
+        echo "${{ contains(steps.get_head_commit_message.outputs.HEAD_COMMIT_MESSAGE, '[generate bindings]') }}"
+  
+  pr_generated_bindings:
+    name: Make PR with generated bindings
+    needs: [test_with_bindgen, check_generate_bindings_flag]
+    if: ${{ contains(needs.check_generate_bindings_flag.outputs.head_commit_message, '[generate bindings]') }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.ref }}
+    - uses: actions/download-artifact@v3
+
+    - name: Update bindings
+      run: |
+        # Update or add the bindings
+        cp generated_binding-*/*.rs bindings/
+
+        # Replace the default bindings
+        cd bindings
+        for x in linux-aarch64 linux-x86_64 macos-aarch64 macos-x86_64 windows-x86_64; do
```

Comment:
Should this be arch-specific? I mean CI/CD runner os/arch specific?

Reply:
don't know. I copied what happens in the other task.
The thing is we might run more binding generations, than we want to keep. And they all will be dumped the same place. And this has to happen, otherwise there are no bindings to commit in the following workflow....

Reply:
I mean, if this is a `linux-x86_64` runner, do we need to copy `windows-x86_64` bindings? At least I read it so here.

Reply:
But it is from the previous workflows, which were done with the right platform. The download-artefact downloads what was uploaded in the previous workflows.

Reply:
> I mean, if this is a `linux-x86_64` runner, do we need to copy `windows-x86_64` bindings?

No. Committing the bindings updates should be done on a single runner. Otherwise there would be about 10 different commits for each platform and cause lots of conflicts.

> The thing is we might run more binding generations, than we want to keep

No. Thanks to `emit-bindings`, the binding generations happen only on the runners that you want to keep. This step is to update the default bindings for each platform.